packages/block-library/src/navigation/use-template-part-area.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * WordPress dependencies
+ */
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+
+// TODO: this util should perhaps be refactored somewhere like core-data.
+import { createTemplatePartId } from '../template-part/edit/utils/create-template-part-id';
+
+export default function useTemplatePartArea( clientId ) {
```

Comment:
This is mostly copied over from `useTemplatePartAreaLabel`, there's a lot of deduplication that could be done here.